ion-hash/src/lib.rs

```diff
@@ -16,15 +16,30 @@
 //! # }
 //! ```
 
+use ion_rs::result::{illegal_operation, IonResult};
 use ion_rs::{value::Element, IonType};
 
 use sha2::{digest::Output, Digest, Sha256};
 
+// A `try`-like macro to workaround the Element API ergonomics. This API
+// requires checking the type and then calling the appropriate getter function
+// (which returns a None if you got it wrong). This macro turns the `None` into
+// an `IonError`.
+// TODO: Remove this once the Element API is reworked.
+macro_rules! t {
```

Comment:
Nit, I really don't like this name, how about `try_to` or some such (though @zslayton doesn't like this name either).  That said, we can refactor this when I merge in my `Dumper` API which steals this definition anyhow.

Reply:
Ack